### PR TITLE
GSoC - Add Post (Picking image from gallery / camera)

### LINKED ISSUE #833: [Bug] Organization Setting not showing up
**Describe the bug**
I created two organisations with the name "A1" and "A3" (Refer to Screenshot 1 - see the creator name). The code should show the 'Organisation setting' on the "Profile" page for those organisations since I am the creator of those organisations but it does not show instead the "Leave Organisation" option is being displayed. (Screenshot 2)

**To Reproduce**

1. Create an organization. (so that you are the creator of the organisation which is being tested)
2. Go to the Profile page (Bottom right button of Bottom navigation bar)
3.  "Leave Organsation" setting is displayed.

**Expected behavior**
Instead of the "Leave Organisation", the "Organsation Setting" option should be displayed.

**Actual behavior**
The "Leave Organisation" option is available which leads to unexpected behavior. (Screenshot 2)

**Screenshots**
Screenshot 1: 
Screenshot 2:

### LINKED ISSUE #883: [GSoC] Tracking the progress of new Add Post Page
### This issue is created to keep track of the major changes needed to be done in redesigning of the **Add Post Page**.

drawing

To do for Add Post Page:

- [x] Creating the initial UI.
- [x] Add uploading file service.
- [ ] Add "Post" service with MVVM.
- [ ] Unit testing of services.

### PR DESCRIPTION
**Pull request by rutvik11062000**
**What kind of change does this PR introduce?**

Added service to take photos from gallery / camera.
Refactored the previous Organization feed loading.

**Issue Number:**

Fixes #883  (2/4 todo)

**Did you add tests for your changes?**
Yes.

**Snapshots/Videos:**

https://user-images.githubusercontent.com/47498552/124352801-5a7ae600-dc20-11eb-93bf-a1ca47016725.mp4

**If relevant, did you update the documentation?**
No.

**Summary**
Checked out 2/4 Todo of the issue.

**Does this PR introduce a breaking change?**
No.

**Other information**
All the third party plugin usage should be abstracted to a different UNIT / service, so that it can be easily replace if needed.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Review by DeltaHarbinger — DISMISSED** [review]
If no one else sees any issues, this may be merged.

File: lib/services/third_party_service/multi_media_pick_service.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  //Local Variables
+  final StreamController<File> _fileStreamController = StreamController();
+  late Stream<File> _fileStream;
+  late ImagePicker _picker;
+
+  //Getters
+  Stream get fileStream => _fileStream;
+
+  Future getPhotoFromGallery({bool camera = false}) async {
```
Comment (TheHazeEffect) [inline lib/services/third_party_service/multi_media_pick_service.dart:24]: return Future<File>

File: lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    _selectedOrg = locator<UserConfig>().currentOrg;
+    _imageFile = null;
+    _multiMediaPickerService = locator<MultiMediaPickerService>();
+  }
+
+  Future getImageFromGallery({bool camera = false}) async {
+    final _image =
+        await _multiMediaPickerService.getPhotoFromGallery(camera: camera);
+    if (_image != null) {
+      _imageFile = _image as File;
```
Comment (TheHazeEffect) [inline lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart:41]: When you return future< file> you'll no longer have to enforce type casting here

File: lib/services/third_party_service/multi_media_pick_service.dart (hunk 1 above)
Comment (rutvik11062000) [reply lib/services/third_party_service/multi_media_pick_service.dart:24]: There can be a condition where we're not able to select a file, in that case we need to return "null" to the viewModel, 
If I keep return type as File I'll not be able to return null from this function.

Comment (TheHazeEffect) [reply lib/services/third_party_service/multi_media_pick_service.dart:24]: Oh this is null safe? return 'Future<File?>' should work

File: lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart (hunk 2 above)
Comment (rutvik11062000) [reply lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart]: Done Please check the latest commit.

**Review by TheHazeEffect — CHANGES_REQUESTED** [review]
Change return type to void from future

File: lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  void initialise() {
+    _currentUser = locator<UserConfig>().currentUser;
+    _navigationService = locator<NavigationService>();
+    _selectedOrg = locator<UserConfig>().currentOrg;
+    _imageFile = null;
+    _multiMediaPickerService = locator<MultiMediaPickerService>();
+  }
+
+  Future getImageFromGallery({bool camera = false}) async {
```
[the remark this answers is not in the document]
Comment (rutvik11062000) [reply lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart]: Please approve if there is no more changes.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**